packages/charts/src/utils/aggregateData.js

```diff
@@ -87,13 +93,16 @@ function aggregate(option, data, unique = true) {
 export default function aggregateData(option, data, unique) {
   const isGroups = Array.isArray(data[0]);
   if (isGroups) {
-    return aggregate(
-      option.split(':')[0],
-      data
-        .map(gd => aggregate(option, gd, unique))
-        .reduce((merge, gd) => merge.concat(gd), []),
-      unique
-    );
+    const aggregatedGroup = data.map(gd => aggregate(option, gd, unique));
+    const func = (option || '').split(':')[0];
+    if (func) {
```

Comment:
`aggregatedGroup` should move inside this `if` since that's the only place I see it being used in this function.

Reply:
Groups are aggregated twice.
Its out of the if because someone can just send `:percent` so no function but convert the data to percent data.